magic_bridge/ic/src/dip20_proxy/src/common/cap.rs

```diff
@@ -0,0 +1,35 @@
+use cap_sdk::{insert, IndefiniteEvent};
+
+use crate::proxy::{ToEvent, STATE};
+
+use super::types::ClaimableMessage;
+
+pub async fn insert_claimable_asset(message: ClaimableMessage) {
+    // add message to STATE claimable messages queue.
+    STATE.with(|s| {
+        s.add_claimable_message(message.clone());
+    });
+
+    // flush STATE claimable messages queue
+    register_messages().await
```

Comment:
Similarly, if we use [cap insert sync](https://github.com/Psychedelic/cap/blob/main/sdk/rust/src/transaction/insert.rs#L126) it handles the flushing mechanism